cpp/ps0004/cpp_ps0004_solution.cpp

```diff
@@ -11,29 +12,26 @@ int main()
     int point = num;
     char ascii = num;
 
-    if (num == 0 || num == 'x')
+    // checking invalidaty of user's input 3/4
+	// printing the number and the opposite printable character 2/2
+    if (num == 0)
     {
         cout << "Invalid character/s or zero value.\n";
-
     }
     else if (num != point)
     {
         cout << "Invalid: a floating point number.\n";
-
     }
 
-    else if (num < 0 || num <= 32 && num >= 126)
+    else if (num < 32 || num > 126)
     {
         cout << "Invalid: number is out of the range.\n";
-
     }
     else
     {
         cout << "the character for  " << num << " is " << ascii << endl;
     }
+    

```

Comment:
- [x] used `if` structure in a right way
- [ ] `if` structure take in consideration all invalid cases
- [ ] all expressions to check invalidity are correct
- [ ] used `std::cout` and `std::endl` or `\n` to print a different message for the 3 cases
- [x] converted the number to the opposite printable character
- [x] printed the number and the opposite printable character